pkg/registry/populator.go

```diff
@@ -373,18 +373,24 @@ type bundleVersion struct {
 // compare returns a value less than one if the receiver arg is less smaller the given version, greater than one if it is larger, and zero if they are equal.
 // This comparison follows typical semver precedence rules, with one addition: whenever two versions are equal with the exception of their build-ids, the build-ids are compared using prerelease precedence rules. Further, versions with no build-id are always less than versions with build-ids; e.g. 1.0.0 < 1.0.0+1.
 func (b bundleVersion) compare(v bundleVersion) (int, error) {
-	if c := b.version.Compare(v.version); c != 0 {
+	return BuildIdCompare(b.version, v.version)
+}
+
+// BuildIdCompare compares two versions and returns negative one if the first arg is less than the second arg, positive one if it is larger, and zero if they are equal.
+// This comparison follows typical semver precedence rules, with one addition: whenever two versions are equal with the exception of their build-ids, the build-ids are compared using prerelease precedence rules. Further, versions with no build-id are always less than versions with build-ids; e.g. 1.0.0 < 1.0.0+1.
+func BuildIdCompare(b semver.Version, v semver.Version) (int, error) {
```

Comment:
If we're going to expose this comparison and use it across multiple packages, we should give it a package of its own.